### PR TITLE
app-crypt/pkcrack: Fix return with no value, in function

### PR DESCRIPTION
**Pull request by listout**
And update EAPI 7 -> 8

Closes: https://bugs.gentoo.org/924227

### CODE REVIEW

File: app-crypt/pkcrack/pkcrack-1.2.2-r2.ebuild [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	doc/README
+)
+
+PATCHES=(
+	"${FILESDIR}/${P}-build.patch"
+	"${FILESDIR}/${P}-gcc14-build.patch"
+)
+
+src_compile() {
+	cd src
```
Comment (juippis) [inline app-crypt/pkcrack/pkcrack-1.2.2-r2.ebuild:37]:
```suggestion
	cd src || die
```

File: app-crypt/pkcrack/pkcrack-1.2.2-r2.ebuild [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	emake CC="$(tc-getCC)" CFLAGS="${CFLAGS} ${LDFLAGS}" all
+}
+
+src_test() {
+	cd test
```
Comment (juippis) [inline app-crypt/pkcrack/pkcrack-1.2.2-r2.ebuild:42]:
```suggestion
	cd test || die
```

File: app-crypt/pkcrack/pkcrack-1.2.2-r2.ebuild [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	emake CC="$(tc-getCC)" CFLAGS="${CFLAGS} ${LDFLAGS}" all
+}
+
+src_install() {
+	einstalldocs
+	cd src
```
Comment (juippis) [inline app-crypt/pkcrack/pkcrack-1.2.2-r2.ebuild:48]:
```suggestion
	cd src || die
```

**Review by juippis — APPROVED** [review]
```
 * QA Notice: One or more compressed files were found in docompress-ed
 * directories. Please fix the ebuild not to install compressed files
 * (manpages, documentation) when automatic compression is used:
 * 
 *   /usr/share/doc/pkcrack-1.2.2-r2/pkzip.ps.gz
```

Test phase fails to me, but seems to work otherwise.

File: app-crypt/pkcrack/pkcrack-1.2.2-r2.ebuild [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+)
+
+PATCHES=(
+	"${FILESDIR}/${P}-build.patch"
+	"${FILESDIR}/${P}-gcc14-build.patch"
+)
+
+src_compile() {
+	cd src
+	emake CC="$(tc-getCC)" CFLAGS="${CFLAGS} ${LDFLAGS}" all
```
Comment (thesamesam) [inline app-crypt/pkcrack/pkcrack-1.2.2-r2.ebuild:38]: emake -C src in both this & src_test lets you drop the `cd src`, but I wonder if S should just be set to include src, given every phase has to do the `cd`..

Comment (juippis) [reply app-crypt/pkcrack/pkcrack-1.2.2-r2.ebuild:38]: I was thinking about it too, but didn't catch whether there's `src_configure` going on and I don't remember if `einstalldocs` worked either outside `${S}`

**Pull request closed without merging**